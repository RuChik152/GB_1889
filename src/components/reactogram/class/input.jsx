import React, { PureComponent } from "react";

export class Input extends PureComponent {

    action = () => {
        this.props.click();
    }

    render() {
        return <div className = "text">
            <input className = "text__input" value = { this.props.text } onChange = { this.props.change } type = "text" placeholder = "Ваше сообщение" name = "msg" />
            <br />

            <button className = "text__btn" onClick = { this.action }>Отправить</button>
        </div>
    }

}