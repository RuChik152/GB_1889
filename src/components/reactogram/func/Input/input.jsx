import React from 'react';

export const Input = ({ text, change, click }) => {
  return (
    <div className="text">
      <input
        className="text__input"
        value={text}
        onChange={change}
        type="text"
        placeholder="Ваше сообщение"
        name="msg"
      />
      <br />
      <button className="text__btn" onClick={click} data-testid="test-id">
        Отправить
      </button>
    </div>
  );
};
