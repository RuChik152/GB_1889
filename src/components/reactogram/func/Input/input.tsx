import React, { FC, FormEvent, SetStateAction, useState} from 'react';
import style from './input.module.scss';
import { Button as ButtonUI, Input as InputUI } from '@mui/material';
import { useDispatch } from 'react-redux';
import { addMsg } from '../store/chats/action';
import { useParams } from 'react-router-dom';

export const Input: FC = () => {
  const [value, setValue] = useState('');
  const { chaiId } = useParams();
  const dispatch = useDispatch();

  const actionForm = (e: FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    if (chaiId) {
      dispatch(addMsg(chaiId, value));
    }
    setValue('');
  };
  return (
    <div className={style.text}>
      <form onSubmit={actionForm}>
        <InputUI
          id="text__input__id"
          value={value}
          onChange={(e) => setValue(e.target.value)}
          type="text"
          placeholder="Ваше сообщение"
        />
        <br />
        <ButtonUI
          disabled={!value}
          data-testid="test-id"
          variant="contained"
        >
          Нажмите Enter
        </ButtonUI>
      </form>
    </div>
  );
};
