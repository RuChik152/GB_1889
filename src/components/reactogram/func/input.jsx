import React, { useEffect, useRef } from 'react';

export const Input = (props) => {
  const inputEl = useRef(null);

  useEffect(() => {
    inputEl.current.focus();
  }, []);

  const action = () => {
    props.click();
  };

  return (
    <div className="text">
      <input
        className="text__input"
        value={props.text}
        onChange={props.change}
        type="text"
        placeholder="Ваше сообщение"
        name="msg"
        ref={inputEl}
      />
      <br />
      <button className="text__btn" onClick={action}>
        Отправить
      </button>
    </div>
  );
};
