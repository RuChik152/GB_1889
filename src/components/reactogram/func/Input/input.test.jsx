import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Input } from './input';
import userEvent from '@testing-library/user-event';

const user = userEvent.setup();

describe('Input', () => {
  it('Render', () => {
    render(
      <Input
        text={'text'}
        author={'admin'}
        change={jest.fn()}
        click={jest.fn()}
      />
    );
  });

  it('is there an INPUT and BUTTON', () => {
    const onSearchMock = jest.fn();
    render(<Input change={onSearchMock} text={'1111'} />);
    expect(screen.getByPlaceholderText('Ваше сообщение')).toBeInTheDocument();
    expect(screen.getByTestId('test-id')).toBeInTheDocument();
  });

  it('Snapshot', () => {
    const { asFragment } = render(
      <Input
        author={'admin'}
        text={'text'}
        change={jest.fn()}
        click={jest.fn()}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('input value to string', () => {
    render(<Input change={jest.fn()} text={'text'} />);
    expect(screen.getByPlaceholderText('Ваше сообщение').value).toMatch(
      String()
    );
  });

  it('button 3 click', async () => {
    const func = jest.fn();
    render(<Input change={jest.fn()} text={'text'} click={func} />);
    await user.click(screen.getByTestId('test-id'));
    expect(func).toHaveBeenCalledTimes(1);
  });

  it('returning data', () => {
    render(<Input />);
    const item = screen.getByTestId('test-id')
    item.value = 'test';
    expect(item.value).toBe('test');
  });
});
