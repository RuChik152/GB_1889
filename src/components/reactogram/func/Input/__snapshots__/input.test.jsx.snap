// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot 1`] = `
<DocumentFragment>
  <div
    class="text"
  >
    <input
      class="text__input"
      name="msg"
      placeholder="Ваше сообщение"
      type="text"
      value="text"
    />
    <br />
    <button
      class="text__btn"
      data-testid="test-id"
    >
      Отправить
    </button>
  </div>
</DocumentFragment>
`;
